File: src/0951.flip-equivalent-binary-trees.988/solution.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root1
 * @param {TreeNode} root2
 * @return {boolean}
 */
var flipEquiv = function (root1, root2) {
    if (root1 && root2) {
        // 树 日常递归判断 考虑交换和不交换的情况
        return root1.val === root2.val && ((flipEquiv(root1.left, root2.left) && flipEquiv(root1.right, root2.right)) || (flipEquiv(root1.left, root2.right) && flipEquiv(root1.right, root2.left)));
    } else if (!root1 && !root2) {
        return true;
    }
    return false;
};

File: src/0951.flip-equivalent-binary-trees.988/solution.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public boolean flipEquiv(TreeNode root1, TreeNode root2) {
        if(root1 == null && root2 == null){
            return true;
        }
        if(root1 == null || root2 == null || root1.val != root2.val){
            return false;
        }
        
        return (flipEquiv(root1.left,root2.left) && flipEquiv(root1.right,root2.right) ) || (  flipEquiv(root1.left,root2.right) && flipEquiv(root1.right,root2.left)   );
    }
}